common/zanata-adapter-glossary/src/main/java/org/zanata/adapter/glossary/GlossaryJsonReader.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2017, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.zanata.adapter.glossary;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.json.JSONException;
+import org.zanata.common.LocaleId;
+import org.zanata.rest.dto.GlossaryEntry;
+import org.json.JSONObject;
+import org.json.JSONArray;
+import org.zanata.rest.dto.GlossaryTerm;
+import org.zanata.rest.dto.QualifiedName;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import static org.apache.commons.lang3.ObjectUtils.firstNonNull;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+/**
+ * @author Damian Jansen <a href="mailto:djansen@redhat.com">djansen@redhat.com</a>
+ **/
+public class GlossaryJsonReader {
+    private final LocaleId srcLang;
+
+    private final static String TERM = "term";
+    private final static String TRANSLATIONS = "translations";
+    private final static String[] POSSYNONYMS = {"pos", "partofspeech", "part of speech"};
```

Comment:
Is this convenience greatly unnecessary?

Reply:
Not really if we want to make it "user friendly". But we will need to clearly state what the system recognised as the part of speech title.